### PR TITLE
ADAPT-1797 Container, Grid, FlexBox components

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY

# Summary
- Add Container, Grid, FlexBox components
- Container has HTML element and width options
- Grid has options for number of columns at each breakpoint, and whether to display grid gap.
- FlexBox has options for flex direction, flex wrap, gaps, and all the different types of box alignments.
- Replace deprecated `babel-eslint `with `@babel/eslist-parser` and update `.eslintrc`

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Look at preview and play with the controls
1. Look at code

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
@sherakama I'll be adding more stuff (flex container, flex cell, grid cell) but please take a look at the preview build when you have time and let me know if you like this approach. The idea is that the `Container` component only controls the width/HTML element and it will wrap around a CSS Grid/Flexbox component.

File: .eslintrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "parser": "babel-eslint",
+  "parser": "@babel/eslint-parser",
```
Comment (yvonnetangsu) [inline .eslintrc:2]: @sherakama eslint wasn't working in my PhpStorm. I found that babel-eslint is deprecated so I updated to this new one and now it's working for me.

**Comment by yvonnetangsu** [comment]
@sherakama I think this one is ready for you to take a look when you have time. Will add the GridCell and FlexCell in a different PR since this one is getting big.

File: src/Alert/Alert.js [hunk 2]
```diff
@@ -15,72 +15,72 @@ import clsxd from 'clsx-dedupe';
  */
 export const Alert = ({ classes = {}, children, ref, ...props }) => {
```
Comment (yvonnetangsu) [inline src/Alert/Alert.js:16]: No real changes to any Alert files except styles linting

File: src/Grid/Grid.levers.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Grid gap or not
+ */
+export const gridGap = [true, false];
+
+/**
+ * Available number of grid columns at XS breakpoint (1 to 12)
+ */
+export const gridColsXs = Array.from({length: 12}, (_, i) => i + 1);
```
Comment (sherakama) [inline src/Grid/Grid.levers.js:9]: Grid cols will always be the same for each breakpoint, won't they? If yes, each of these breakpoint col variables could be consolidated down into one.

File: src/Grid/Grid.stories.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  component: Grid,
+  argTypes: {
+    xs: {
+      control: {
+        type: 'range',
+        min: 1,
+        max: gridColsXs[gridColsXs.length-1],
+        step: 1
+      }
+    },
```
Comment (sherakama) [inline src/Grid/Grid.stories.js:32]: This is niiiice.

File: src/Grid/Grid.levers.js (hunk 3 above)
Comment (yvonnetangsu) [reply src/Grid/Grid.levers.js:9]: Originally I played with having just 1-6 for XS but I changed my mind 🤔

**Review by sherakama — APPROVED** [review]
One small change to consolidate the grid column variable and then this is GTG.

File: src/Grid/Grid.levers.js (hunk 3 above)
Comment (yvonnetangsu) [reply src/Grid/Grid.levers.js]: All done

**Review by sherakama — APPROVED** [review]
GTG

**Pull request merged**